src/Giraffe/Auth.fs

```diff
@@ -36,6 +37,28 @@ let signOut (authScheme : string) : HttpHandler =
             return! next ctx
         }
 
+/// ** Description **
+/// Validates if the user satisfies a certain policy. If the requirement is satisfied then it will continue with the `next` function otherwise it will shortcircuit to the `authFailedHandler`.
+///
+/// ** Parameters **
+///     - `policy`: A `string` containing the desired policy to validate
+///     - `authFailedHandler`: A `HttpHandler` function which will be executed when the policy is not satisfied.
+///
+/// ** Output **
+/// A Giraffe `HttpHandler` function which can be composed into a bigger web application.
+let requiresPolicy (policy : string) (authFailedHandler : HttpHandler) : HttpHandler =
+    fun (next : HttpFunc) (ctx : HttpContext) ->                
+        match ctx.GetService<IAuthorizationService>() with
+        |null ->                     
+            authFailedHandler finish ctx
```

Comment:
Not sure if we should invoke the `authFailedHandler` when the `IAuthorizationService` hasn't been registered. It might give the user the wrong impression of where the error lies. I think I'd prefer to just let a NRE bubble up, or change the `ctx.GetService<'T>` implementation to check for `null` and throw a meaningful exception in one place for all `GetService` calls. What do you think?

Reply:
Probably use the `GetRequiredService` would be fitting as it already throws on missing service, but yeah, agree with the first part.

Reply:
Looks like we can't use the `GetRequiredService` on that context, so it could be nice as a method extension to throw a fitting exception.

Reply:
Didn't see that `GetService` was already an extension. Should be ok to throw on missing. I'll add something to a future commit.